File: backend/app/schemas/recette_lso.py
"""Schemas Pydantic pour RecetteLSO (Sol Vivant)"""
from pydantic import BaseModel, ConfigDict
from typing import Optional, List


class RecetteLSOLigneBase(BaseModel):
    id_produit: int
    quantite:   float
    unite:      Optional[str] = None   # mL | L | g | Kg
    note_ligne: Optional[str] = None
    ordre:      int = 0


class RecetteLSOLigneCreate(RecetteLSOLigneBase):
    pass


class RecetteLSOLigneRead(RecetteLSOLigneBase):
    id_ligne:     int
    nom_produit:  Optional[str] = None
    type_produit: Optional[str] = None
    model_config  = ConfigDict(from_attributes=True)


class RecetteLSOBase(BaseModel):
    nom_recette:     str
    type_lso:        Optional[str]  = None
    quantite_totale: Optional[float]= None
    unite_quantite:  Optional[str]  = None   # L | Kg
    notes:           Optional[str]  = None


class RecetteLSOCreate(RecetteLSOBase):
    lignes: List[RecetteLSOLigneCreate] = []


class RecetteLSOUpdate(BaseModel):
    nom_recette:     Optional[str]  = None
    type_lso:        Optional[str]  = None
    quantite_totale: Optional[float]= None
    unite_quantite:  Optional[str]  = None
    notes:           Optional[str]  = None
    lignes:          Optional[List[RecetteLSOLigneCreate]] = None


class RecetteLSORead(RecetteLSOBase):
    id_recette_lso: int
    lignes:         List[RecetteLSOLigneRead] = []
    model_config    = ConfigDict(from_attributes=True)

File: frontend/src/api/recetteLSO.ts
import client from './client'

export interface RecetteLSOLigne {
  id_ligne?:    number
  id_produit:   number
  quantite:     number
  unite?:       string   // mL | L | g | Kg
  note_ligne?:  string
  ordre:        number
  nom_produit?: string
  type_produit?: string
}

export interface RecetteLSO {
  id_recette_lso:  number
  nom_recette:     string
  type_lso?:       string
  quantite_totale?: number
  unite_quantite?:  string   // L | Kg
  notes?:          string
  lignes:          RecetteLSOLigne[]
}

export type RecetteLSOCreate = Omit<RecetteLSO, 'id_recette_lso'>
export type RecetteLSOUpdate = Partial<RecetteLSOCreate>

export const recetteLSOAPI = {
  getAll:    ()                                       => client.get<RecetteLSO[]>('/recettes-lso/'),
  getById:   (id: number)                             => client.get<RecetteLSO>(`/recettes-lso/${id}`),
  create:    (data: RecetteLSOCreate)                 => client.post<RecetteLSO>('/recettes-lso/', data),
  update:    (id: number, data: RecetteLSOUpdate)     => client.put<RecetteLSO>(`/recettes-lso/${id}`, data),
  delete:    (id: number)                             => client.delete(`/recettes-lso/${id}`),
  exportCSV: ()                                       => client.get('/recettes-lso/export/csv', { responseType: 'blob' }),
  importCSV: (file: File) => {
    const form = new FormData(); form.append('file', file)
    return client.post('/recettes-lso/import', form, { headers: { 'Content-Type': 'multipart/form-data' } })
  },
}
